### ISSUE TITLE
[array-bracket-spacing] doesn't work on types

### ISSUE THREAD

**Issue by AlbanDurrheimer** [typescript-eslint/typescript-eslint#3105]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "array-bracket-spacing": ["error"]
  }
}
```

```TS
const i: [string ]: ['Hello world'];
```

**Expected Result**

ESLint: There should be no space before ']'.(array-bracket-spacing)  

**Actual Result**

No problems found  

**Additional Info**

Rule `@typescript-eslint/array-bracket-spacing` not found in [the list of "extension" rules](https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/README.md#extension-rules) and does not work with the `canary` tag install.  

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.15.2` |
| `@typescript-eslint/parser`        | `4.15.2` |
| `TypeScript`                       | `4.1.5` |
| `ESLint`                           | `7.20.0` |
| `node`                             | `14.16.0` |

**Comment by muzmich** [comment]
+1

**Comment by JoshuaKGoldberg** [comment]
@muzmich please don't post +1. It doesn't help get the issue done faster. It just spams us volunteer maintainers and makes us crabby 🙂. If you want to upvote an issue, use the 👍 reaction on it.

**Comment by Josh-Cena** [comment]
Because the ESLint core is also working on [deprecating/removing its formatting rules](https://github.com/eslint/eslint/issues/17522), we will close this. A 3rd-party plugin providing formatting, like https://github.com/eslint-stylistic/eslint-stylistic, can surely support TS OOTB.
